src/main/java/org/acm/auth/managers/ReactionManager.java

```diff
@@ -0,0 +1,40 @@
+package org.acm.auth.managers;
+
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.channel.priv.PrivateChannelCreateEvent;
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent;
+import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.acm.auth.utils.ReactionRoles;
+import org.jetbrains.annotations.NotNull;
+
+public class ReactionManager extends ListenerAdapter {
+
+    @Override
+    public void onGuildMessageReactionAdd(@NotNull GuildMessageReactionAddEvent event) {
+        if (event.getUser().isBot())
+            return;
+
+        Role role = ReactionRoles.getReactionRole(event.getMessageId(), event.getReactionEmote().getEmoji());
+        event.getGuild().addRoleToMember(event.getMember(), role).queue();
+
+        //sends pm to user for added role
+        event.getUser().openPrivateChannel()
+                .flatMap(privateChannel -> privateChannel.sendMessage("You got role **" + role.getName() + "**"))
+                .queue();
+
+        ReactionRoles.addUser(event.getUserId(), event.getUser());
```

Comment:
Why do we need this?

Reply:
On "onMessageReactionRemove" event.getUser() produced NullPointerException. I searched a liitle and I think that some users weren't cached.

Reply:
Seems like you're right. However, according to [this StackOverflow post](https://stackoverflow.com/a/63945376), the right way to approach this would be to get the user's ID and call the remove role method that takes an String instead of a User.

Reply:
The reason I wanted User was to send a private confirmation message. If we don't need that feature it's not necessary to save users.

Reply:
You're looking for [this](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/JDA.html#openPrivateChannelById(java.lang.String)) then :)

Reply:
Perfect. Thanks!